tools/log-collector/log_collector.go

```diff
@@ -797,3 +810,24 @@ func (l *LogCollector) checkNamespaces() error {
 
 	return nil
 }
+
+// getApiGVResource returns resource for given resourceName
+func (l *LogCollector) getAPIGVResource(apiGroupVersion, resourceName string) (gVResources apiv1.APIResource, err error) {
+
+	var gVResourcesList *apiv1.APIResourceList
+	gVResourcesList, err = l.disClient.ServerResourcesForGroupVersion(apiGroupVersion)
```

Comment:
You'll have make `disClient` calls for same group multiples times if different `resourceName` is passed to func `getAPIGVResource`. Make one  `disClient` call for one group and add it map and use it later for different `resourceName` [no need to call func `getApiGVResource` everytime].

Reply:
`getAPIGVResource` and `getAPIGVResources`  are two different for group and specific resource